### PR TITLE
Added option to remove all CRs from input stream

### PR DESCRIPTION
**Pull request by butitsnotme**
### Required for all PRs:

- [x] CHANGELOG.md updated
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

Added the option `removecr` to `inputs.exec` to remove all carriage returns
(CR, ASCII 0x0D, Unicode codepoint \u0D, ^M). The option is boolean and
not enabled if not present in the config file.

To be clear it removes them from the output of the command before it is processed by the parser.

### CODE REVIEW

**Comment by butitsnotme** [comment]
I'm not sure why the test failed, it is the TestCombinedOutputTimeout test that is failing. It runs fine on my test machine (Windows) (though many other tests fail because they're expecting services...).

Perhaps the test just need to be re-run? (Or maybe my change affects this, but don't see anything I changed getting called)

**Comment by sparrc** [comment]
Why did you make it a configurable option? is there ever a reason someone would want to keep the carriage returns?

**Comment by butitsnotme** [comment]
I can't think of a reason, should I remove the option?

**Comment by sparrc** [comment]
yes, please do, but possibly we should be doing this at a lower level in the command running exec functions: https://github.com/influxdata/telegraf/blob/master/internal/internal.go#L168

and it should also only happen `if runtime.GOOS == "windows"`

File: internal/internal.go [hunk 1]
```diff
@@ -172,7 +173,29 @@ func CombinedOutputTimeout(c *exec.Cmd, timeout time.Duration) ([]byte, error) {
 	if err := c.Start(); err != nil {
 		return nil, err
 	}
+
 	err := WaitTimeout(c, timeout)
+
+	if runtime.GOOS == "windows" {
+		var buf bytes.Buffer
```
Comment (sparrc) [inline internal/internal.go]: put this logic within the if-statement in it's own function and unit test and put a comment above the function explaining what it's doing.

**Comment by butitsnotme** [comment]
@sparrc inputs.exec does not call internal.CombinedOutputTimeout(), it calls exec.Command() directly, only using internal for internal.RunTImeout().

Should it be using internal.CombinedOutputTimeout()? (Currently it does not include stderr in the output.)

**Comment by sparrc** [comment]
yes, I think I was wrong, we should probably just strip the CRs from the exec plugin only, as you had it before.

**Comment by sparrc** [comment]
looks good @butitsnotme, but you need to fix your CHANGELOG update. I would suggest just throwing away your changes to the changelog and rebasing, then adding your PR to the list again.

**Comment by steverweber** [comment]
+1 would like this in the RC

**Comment by butitsnotme** [comment]
I've updated the changelog and rebased on master. I've tested the latest commit on Windows, it all works. The failed check appears to be unrelated (it looks like one of the docker images didn't initialize in time).

I would very much appreciate it if this could be incorporated into the 1.0 release, as without it the exec plugin is nearly useless on Windows.

**Comment by butitsnotme** [comment]
The failed check looks like it has nothing to do with this... (Aerospike failed)

**Comment by sparrc** [comment]
thx @butitsnotme

**Pull request closed without merging**